minetest/minetest
# [NO SQ] Settings: semi-automatic callback cleanup

First commit: cleanup by userdata
Second commit: header cleanups

## To do

This PR is Ready for Review.

## How to test

1. It must compile
2. Add logging to `deregisterAllChangedCallbacks` to ensure there's always >= 1 registration removed.
